TruePath/PathExtensions.CombineWith.cs

```diff
@@ -0,0 +1,32 @@
+namespace TruePath;
+
+public static partial class PathExtensions
+{
+    public static IPath CombineWith(this IPath path, IPath other)
+    {
+        var combined = Path.Combine(path.Value, other.Value);
+
+        return Path.IsPathFullyQualified(combined) ? new AbsolutePath(combined) : new LocalPath(combined);
+    }
+
+    public static IPath CombineWith(this IPath path, string other)
+    {
+        var combined = Path.Combine(path.Value, other);
+
+        return Path.IsPathFullyQualified(combined) ? new AbsolutePath(combined) : new LocalPath(combined);
+    }
+
+    public static IPath CombineWithAsPath(this string path, string other)
+    {
+        var combined = Path.Combine(path, other);
+
+        return Path.IsPathFullyQualified(combined) ? new AbsolutePath(combined) : new LocalPath(combined);
+    }
+
+    public static IPath CombineWithAsPath(this string path, IPath other)
+    {
+        var combined = Path.Combine(path, other.Value);
+
+        return Path.IsPathFullyQualified(combined) ? new AbsolutePath(combined) : new LocalPath(combined);
+    }
```

Comment:
I do not understand why these methods are useful. Moreover, they defy the type safety, and the resulting `IPath` is mostly useless: you cannot do much to it without force-casting to one of the concrete types.

Why is this required? What are the use cases?

For what I see, you should be able to do something like `new LocalPath(yourPath) / other` for the most of these, and get more useful result.